tools/workflow/cleaner/ecs/clean.go

```diff
@@ -0,0 +1,137 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package ecs
+
+import (
+	"fmt"
+	"log"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/ecs"
+)
+
+const (
+	Type          = "ecs"
+	clusterPrefix = "aoc-testing"
+	taskDefPrefix = "taskdef"
+)
+
+var logger = log.New(os.Stdout, fmt.Sprintf("[%s] ", Type), log.LstdFlags)
+
+func Clean(sess *session.Session, expirationDate time.Time) error {
+	logger.Printf("Begin to clean ECS Task Definitions")
+	client := ecs.New(sess)
+
+	var nextToken *string
+	for {
+		lci := &ecs.ListClustersInput{NextToken: nextToken}
+		lco, err := client.ListClusters(lci)
+		if err != nil {
+			return err
+		}
+		dci := &ecs.DescribeClustersInput{Clusters: lco.ClusterArns}
+		dco, err := client.DescribeClusters(dci)
+		if err != nil {
+			return err
+		}
+		for _, cluster := range dco.Clusters {
+			if strings.HasPrefix(*cluster.ClusterName, clusterPrefix) {
+				if *cluster.ActiveServicesCount > 0 {
+					allServicesDeleted, err := deleteServices(client, expirationDate, cluster.ClusterArn)
+					if err != nil {
+						return err
```

Comment:
Is returning an err here the desirable behavior? Wouldn't we want to iterate through all clusters even if one attempted deletion failed? Is there a specific scenario where we want to abort when one cluster fails to delete?